src/time.cc

```diff
@@ -118,43 +126,45 @@ class Time {
+      (obj).Set(Napi::String::New(env, "timezone"), Napi::String::New(env, tzname[timeinfo->tm_isdst]));
 #endif // HAVE_TM_GMTOFF
     } else {
-      Nan::Set(obj, Nan::New("invalid").ToLocalChecked(), Nan::New<v8::Boolean>(true));
+      (obj).Set(Napi::String::New(env, "invalid"), Napi::Boolean::New(env, true));
     }
 
-    info.GetReturnValue().Set(scope.Escape(obj));
+    return scope.Escape(obj);
   }
 
-  static NAN_METHOD(Mktime) {
-    Nan::EscapableHandleScope scope;
+  static Napi::Value Mktime(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    Napi::EscapableHandleScope scope(env);
 
     if (info.Length() < 1) {
-      return Nan::ThrowTypeError("localtime() Object expected");
+      Napi::TypeError::New(env, "localtime() Object expected").ThrowAsJavaScriptException();
+      return env.Null();
     }
 
-    Local<Object> arg = info[0].As<v8::Object>();
+    Napi::Object arg = info[0].As<Napi::Object>();
 
     struct tm tmstr;
-    tmstr.tm_sec   = Nan::Get(arg, Nan::New("seconds").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_min   = Nan::Get(arg, Nan::New("minutes").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_hour  = Nan::Get(arg, Nan::New("hours").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_mday  = Nan::Get(arg, Nan::New("dayOfMonth").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_mon   = Nan::Get(arg, Nan::New("month").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_year  = Nan::Get(arg, Nan::New("year").ToLocalChecked()).ToLocalChecked()->Int32Value();
-    tmstr.tm_isdst = Nan::Get(arg, Nan::New("isDaylightSavings").ToLocalChecked()).ToLocalChecked()->Int32Value();
+    tmstr.tm_sec   = (arg).Get(Napi::String::New(env, "seconds")).As<Napi::Number>().Int32Value();
+    tmstr.tm_min   = (arg).Get(Napi::String::New(env, "minutes")).As<Napi::Number>().Int32Value();
+    tmstr.tm_hour  = (arg).Get(Napi::String::New(env, "hours")).As<Napi::Number>().Int32Value();
+    tmstr.tm_mday  = (arg).Get(Napi::String::New(env, "dayOfMonth")).As<Napi::Number>().Int32Value();
+    tmstr.tm_mon   = (arg).Get(Napi::String::New(env, "month")).As<Napi::Number>().Int32Value();
+    tmstr.tm_year  = (arg).Get(Napi::String::New(env, "year")).As<Napi::Number>().Int32Value();
+    tmstr.tm_isdst = (arg).Get(Napi::String::New(env, "isDaylightSavings")).As<Napi::Number>().Int32Value();
     // tm_wday and tm_yday are ignored for input, but properly set after 'mktime' is called
 
-    info.GetReturnValue().Set(scope.Escape(Nan::New<v8::Number>(static_cast<double>(mktime( &tmstr )))));
+    return scope.Escape(Napi::Number::New(env, static_cast<double>(mktime( &tmstr ))));
   }
 
 };
 
 extern "C" {
-  static void init (Handle<Object> target) {
-    Time::Init(target);
+  static Napi::Object init (Napi::Env env, Napi::Object exports) {
+    return Time::Init(env, exports);
   }
-  NODE_MODULE(time, init)
+  NODE_API_MODULE(time, init)
 }
```

Comment:
I'm not sure that we need this `extern` block any more, or the wrapper around `Time::Init`

```cpp
NODE_API_MODULE(time, Time::Init);
```

Reply:
But macro NODE_API_MODULE doesn't work with namespace in regfunc  https://github.com/nodejs/node-addon-api/issues/249